cosmwasm/contracts/outpost/src/execute.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    msg::{Callback, ExecuteMsg, Wasm, WasmHookExecute},
+    state::CONFIG,
+    ContractError,
+};
+use cosmwasm_std::{Addr, Coin, DepsMut, Response, Timestamp};
+
+// IBC timeout
+pub const PACKET_LIFETIME: u64 = 604_800u64; // One week in seconds
+
+//#[cfg(feature = "callbacks")]
+fn build_callback_memo(callback: Option<Callback>) -> Result<String, ContractError> {
+    match callback {
+        Some(callback) => callback.try_string(),
+        None => Ok(String::new()),
+    }
+}
+
+pub fn execute_swap(
+    deps: DepsMut,
+    own_addr: Addr,
+    now: Timestamp,
+    coin: Coin,
+    user_msg: ExecuteMsg,
```

Comment:
Why don't we just type this to `ExecuteMsg::OsmosisSwap` ?

Reply:
I *think* that's not possible (because the type is the enum and OsmosisSwap is just a variant). We have to do the match. But it would be great if we could! 

Happy to change it if you know how

Reply:
yeah: 

```
error[E0573]: expected type, found variant `ExecuteMsg::OsmosisSwap`
  --> contracts/outpost/src/execute.rs:25:15
   |
25 |     user_msg: ExecuteMsg::OsmosisSwap,
   |               ^^^^^^^^^^^^^^^^^^^^^^^ not a type
   |
```